### PR TITLE
Add error handling for failed insert on mongo

### PR DESCRIPTION
**Pull request by peppe-hmrc**

### CODE REVIEW

File: app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala [hunk 1]
```diff
@@ -24,7 +24,8 @@ trait CTCError{
   val message: String
 }
 
-case class MalformedRequest(message: String = "Malformed request") extends CTCError{}
+case class MalformedRequest(message: String = "Malformed request") extends CTCError
+case class MongodbError(message: String) extends CTCError
```
Comment (udinobi) [inline app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala]: I am aware we are using Mongo and that there is no plan to change that, but still I'd like to have for this case class a more generic name, e.g. PersistenceError, StorageError, or something like that. Something not directly related to Mongo, btw.

File: app/uk/gov/hmrc/ctc/handlers/ErrorHandler.scala [hunk 2]
```diff
@@ -19,16 +19,17 @@ package uk.gov.hmrc.ctc.handlers
...
 
 trait ErrorHandler {
 
   def toResult(error: CTCError): Result = {
     error match {
-      case e:MalformedRequest => BadRequest(Json.toJson(ErrorResponse(400,e.message)))
-      case _                  => BadRequest("The request can't be processed")
+      case e:MalformedRequest => BadRequest(Json.toJson(ErrorResponse(400, e.message)))
+      case _:MongodbError => InternalServerError(Json.toJson(ErrorResponse(500, "The request can't be fulfilled at the moment")))
+      case _                  => BadRequest(Json.toJson(ErrorResponse(400, "The request can't be processed")))
```
Comment (udinobi) [inline app/uk/gov/hmrc/ctc/handlers/ErrorHandler.scala]: How can we derive the default case is triggered by a wrong request ?

File: app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala (hunk 1 above)
Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/errors/ErrorResponse.scala]: I agree, however it sound that is more a naming issue ... feel free to open a ticket and work on it. I don't think is strictly related to this pr

File: app/uk/gov/hmrc/ctc/handlers/ErrorHandler.scala (hunk 2 above)
Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/handlers/ErrorHandler.scala]: Well you right, I'm just defaulting it to 400. There are no business requirements on that

Comment (udinobi) [reply app/uk/gov/hmrc/ctc/handlers/ErrorHandler.scala]: A 500 code wouldn't at least say to the client that nothing is wrong from their side ?

Comment (udinobi) [reply app/uk/gov/hmrc/ctc/handlers/ErrorHandler.scala]: Googling I found...
> You should use 4xx for a client error if the client can modify the request to get around the error. Use a 5xx for a server error that the client can't really work around.

Comment (peppe-hmrc) [reply app/uk/gov/hmrc/ctc/handlers/ErrorHandler.scala]: yep I see and I agree with it ... just changes this

**Pull request merged**